src/Raven.Server/Documents/Replication/AllowedPathsValidator.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Collections.Generic;
+using Raven.Server.Documents.Replication.ReplicationItems;
+using Sparrow.Json;
+using Sparrow.Server.Utils;
+using Voron;
+
+namespace Raven.Server.Documents.Replication
+{
+    public class AllowedPathsValidator : IDisposable
+    {    
+        private readonly JsonOperationContext _allowedPathsContext;
+        private readonly List<LazyStringValue> _allowedPaths;
+        private readonly List<LazyStringValue> _allowedPathsPrefixes;
+
+        
+        private unsafe LazyStringValue _tmpLazyStringInstance;
+        private readonly AllowedPathsValidator _allowedPathsValidator;
+
+        private unsafe LazyStringValue GetDocumentId(Slice key)
+        {
+            var sepIdx = key.Content.IndexOf(SpecialChars.RecordSeparator);
+            if(_tmpLazyStringInstance == null)
+                _tmpLazyStringInstance = new LazyStringValue(null, null, 0, _allowedPathsContext);
+            _tmpLazyStringInstance.Renew(null, key.Content.Ptr, sepIdx);
+            return _tmpLazyStringInstance;
+        }
+
+        public string GetItemInformation(ReplicationBatchItem item)
+        {
+            return item switch
+            {
+                AttachmentReplicationItem a => "Attachment for " + GetDocumentId(a.Key),
+                AttachmentTombstoneReplicationItem at => "Attachment tombstone for: " + GetDocumentId(at.Key),
+                CounterReplicationItem c => "Counter for " + c.Id,
+                DocumentReplicationItem d => "Document " + d.Id,
+                RevisionTombstoneReplicationItem r => "Revision for " + r.Id,
+                TimeSeriesDeletedRangeItem td => "Time Series deletion range for: " +  GetDocumentId(td.Key),
+                TimeSeriesReplicationItem t => "Time Series For: " + GetDocumentId(t.Key),
```

Comment:
We have some inconsistency in those strings - ` for: ` vs `for `

Reply:
fixed